pulsar-broker/src/main/java/org/apache/pulsar/broker/transaction/buffer/impl/TopicTransactionBuffer.java

```diff
@@ -219,19 +219,7 @@ public CompletableFuture<Void> checkIfTBRecoverCompletely(boolean isTxnEnabled)
         } else {
             CompletableFuture<Void> completableFuture = new CompletableFuture<>();
             transactionBufferFuture.thenRun(() -> {
-                if (checkIfNoSnapshot()) {
-                    snapshotAbortedTxnProcessor.takeAbortedTxnsSnapshot(maxReadPosition).thenRun(() -> {
```

Comment:
If `isTxnEnabled` is true, there need to take a snapshot and change to ready state.

Reply:
`isTxnEnabled` will be set to true forcefully in `org.apache.pulsar.broker.service.ServerCnx#handleGetLastMessageId`.
https://github.com/apache/pulsar/blob/17e3f860b050443de74413c1d0b4a3d47173f68a/pulsar-broker/src/main/java/org/apache/pulsar/broker/service/ServerCnx.java#L2219-L2219

Reply:
It is possible to optimize this area by taking a snapshot before writing the first transactional message, rather than when the client connects.

Reply:
Make sense, i have updated the code, PTAL, thanks.